substrate/frame/support/src/lib.rs

```diff
@@ -1019,1221 +916,840 @@ pub mod pallet_prelude {
+/// aim of dev mode is to loosen some of the restrictions and requirements placed on
+/// production pallets for easy tinkering and development. Dev mode pallets should not be
+/// used in production. Enabling dev mode has the following effects:
 ///
-/// NOTE: Bypassing the `frame_system::Config` supertrait check is typically desirable when you
-/// want to write an alternative to the `frame_system` pallet.
+/// * Weights no longer need to be specified on every `#[pallet::call]` declaration. By
+///   default, dev mode pallets will assume a weight of zero (`0`) if a weight is not
+///   specified. This is equivalent to specifying `#[weight(0)]` on all calls that do not
+///   specify a weight.
+/// * Call indices no longer need to be specified on every `#[pallet::call]` declaration. By
+///   default, dev mode pallets will assume a call index based on the order of the call.
+/// * All storages are marked as unbounded, meaning you do not need to implement
+///   [`MaxEncodedLen`](frame_support::pallet_prelude::MaxEncodedLen) on storage types. This is
+///   equivalent to specifying `#[pallet::unbounded]` on all storage type definitions.
+/// * Storage hashers no longer need to be specified and can be replaced by `_`. In dev mode,
+///   these will be replaced by `Blake2_128Concat`. In case of explicit key-binding, `Hasher`
+///   can simply be ignored when in `dev_mode`.
 ///
-/// Also see
-/// [`pallet::disable_frame_system_supertrait_check`](`frame_support::pallet_macros::disable_frame_system_supertrait_check`)
+/// Note that the `dev_mode` argument can only be supplied to the `#[pallet]` or
+/// `#[frame_support::pallet]` attribute macro that encloses your pallet module. This
+/// argument cannot be specified anywhere else, including but not limited to the
+/// `#[pallet::pallet]` attribute macro.
 ///
-/// ## Macro expansion:
+/// <div class="example-wrap" style="display:inline-block"><pre class="compile_fail"
+/// style="white-space:normal;font:inherit;">
+/// <strong>WARNING</strong>:
+/// You should not deploy or use dev mode pallets in production. Doing so can break your
+/// chain and therefore should never be done. Once you are done tinkering, you should
+/// remove the 'dev_mode' argument from your #[pallet] declaration and fix any compile
+/// errors before attempting to use your pallet in a production scenario.
+/// </pre></div>
 ///
-/// The macro expands pallet constant metadata with the information given by
-/// `#[pallet::constant]`.
+/// ## Runtime Metadata Documentation
 ///
-/// # `pallet::generate_store($vis trait Store)`
+/// The documentation added to this pallet is included in the runtime metadata.
 ///
-/// To generate a `Store` trait associating all storages, annotate your `Pallet` struct with
-/// the attribute `#[pallet::generate_store($vis trait Store)]`, e.g.:
+/// The documentation can be defined in the following ways:
 ///
 /// ```ignore
```

Comment:
I don't think this code block specifically can be made to compile, because the .md files don't exist?

Reply:
Yeah this is a reasonable exception.